File: src/console/command/impl/TimingsCommand.php
<?php

namespace pocketcloud\cloud\console\command\impl;

use pocketcloud\cloud\console\command\Command;
use pocketcloud\cloud\console\command\sender\ICommandSender;
use pocketcloud\cloud\console\command\SubCommand;
use pocketcloud\cloud\util\benchmark\Benchmark;
use pocketcloud\cloud\util\benchmark\BenchmarkTimingsSummary;
use pocketcloud\cloud\util\PathUtils;
use const pocketcloud\TIMINGS_PATH;

final class TimingsCommand extends Command {

    public function __construct() {
        parent::__construct("timings", "Manage the timings");
        $this->enableUseRegularHandlerForSubCommands();
        $this->registerSubCommand(SubCommand::nonHandler("paste"));
        $this->registerSubCommand(SubCommand::nonHandler("dump"));
    }

    public function run(ICommandSender $sender, string $label, array $args, ?SubCommand $subCommand, array $flags): bool {
        if ($subCommand?->getName() == "paste") {
            $sender->info("pasing timings into §b{}§r...", $path = PathUtils::join(TIMINGS_PATH, date("Y-m-d_H:i:s_T") . ".txt"));
            if (Benchmark::writeTimings($path)) {
                $sender->success("Successfully §apasted §rtimings.");
            } else $sender->warn("Failed to paste timings.");
        } else if ($subCommand?->getName() == "dump") {
            /** @var BenchmarkTimingsSummary $summary */
            foreach (Benchmark::getSummary() as $summary) {
                $sender->info($summary->format());
            }
        }
        return true;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/impl/TimingsCommand.java
package de.pocketcloud.cloud.console.command.impl;

import de.pocketcloud.cloud.console.command.Command;
import de.pocketcloud.cloud.console.command.ctx.CommandContext;
import de.pocketcloud.cloud.console.command.desc.CommandDescription;
import de.pocketcloud.cloud.console.command.sender.CommandSender;
import de.pocketcloud.cloud.console.command.sub.SubCommand;
import de.pocketcloud.cloud.util.PocketCloudPaths;
import de.pocketcloud.cloud.util.benchmark.Benchmark;
import de.pocketcloud.cloud.util.benchmark.BenchmarkTimingsSummary;

import java.nio.file.Path;
import java.text.SimpleDateFormat;
import java.util.Comparator;
import java.util.Date;
import java.util.List;

@CommandDescription(name = "timings", description = "View the cloud's timings")
public final class TimingsCommand extends Command {

    @Override
    public void prepare() {
        registerSubCommand(SubCommand.lambda("paste", this::handlePasteSub));
        registerSubCommand(SubCommand.lambda("dump", this::handleDumpSub));
    }

    @Override
    public boolean run(CommandSender sender, CommandContext ctx) {
        return false;
    }

    public boolean handlePasteSub(CommandSender sender, CommandContext ctx) {
        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
        Path path = PocketCloudPaths.storage().timings().with(formatter.format(new Date()) + ".txt").asPath();
        sender.info("§aPasting §rtimings into §b{}§r...", path.toAbsolutePath().toString());
        if (Benchmark.writeTimings(path, true)) {
            sender.success("Successfully §apasted §rtimings.");
        } else {
            sender.error("Failed to paste timings.");
        }
        return true;
    }

    public boolean handleDumpSub(CommandSender sender, CommandContext ctx) {
        List<BenchmarkTimingsSummary> timings = Benchmark.getSummary(Comparator.comparingDouble(BenchmarkTimingsSummary::max));
        for (BenchmarkTimingsSummary timing : timings) {
            sender.info(timing.format(3, true));
        }
        return true;
    }
}